packages/datadog-instrumentations/src/router.js

```diff
@@ -169,11 +169,105 @@ function createWrapRouterMethod (name) {
 
 const wrapRouterMethod = createWrapRouterMethod('router')
 
-addHook({ name: 'router', versions: ['>=1'] }, Router => {
+addHook({ name: 'router', versions: ['>=1 <2'] }, Router => {
   shimmer.wrap(Router.prototype, 'use', wrapRouterMethod)
   shimmer.wrap(Router.prototype, 'route', wrapRouterMethod)
 
   return Router
 })
 
+const queryParserReadCh = channel('datadog:query:read:finish')
+
+addHook({ name: 'router', versions: ['>=2'] }, Router => {
+  const WrappedRouter = shimmer.wrapFunction(Router, function (originalRouter) {
+    return function wrappedMethod (options) {
+      const router = originalRouter.call(this, options)
```

Comment:
i don't see a reason to use .call() here
```suggestion
    return function wrappedMethod () {
      const router = originalRouter.call(this, arguments)
```

Reply:
Yes we can do it with an apply
```suggestion
    return function wrappedMethod (options) {
      const router = originalRouter.apply(this, arguments)
```

Reply:
you're not using options tho are you ?

Reply:
i have the habit of including the arguments only if they're really relevant to the instrumentation